plugins/graph_algorithm/include/graph_algorithm/algorithms/neighborhood.h

```diff
@@ -0,0 +1,83 @@
+// MIT License
+//
+// Copyright (c) 2019 Ruhr University Bochum, Chair for Embedded Security. All Rights reserved.
+// Copyright (c) 2019 Marc Fyrbiak, Sebastian Wallat, Max Hoffmann ("ORIGINAL AUTHORS"). All rights reserved.
+// Copyright (c) 2021 Max Planck Institute for Security and Privacy. All Rights reserved.
+// Copyright (c) 2021 Jörn Langheinrich, Julian Speith, Nils Albartus, René Walendy, Simon Klix ("ORIGINAL AUTHORS"). All Rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#pragma once
+
+#include "graph_algorithm/netlist_graph.h"
+#include "hal_core/defines.h"
+#include "hal_core/utilities/result.h"
+
+#include <igraph/igraph.h>
```

Comment:
Normally this is `#include <igraph.h>`, and `-Iprefix/include/igraph` is used instead of `-Iprefix/include`.

The CMake and pkgconfig files that igraph installs are all set up to provide these include paths.

Reply:
If possible I'd also rather use `find_package()` and rely on CMake to have include path set. I guess this is what you suggested. The problem is that this mechanism (to my understanding which is very poor) works only for packages already installed. Our goal however is to build the igraph library together with HAL so we can incorporate the latest version. Therefore the files are not in place yet when we run configuration with CMake.

Reply:
> I guess this is what you suggested. 

Yes, indeed.

> Our goal however is to build the igraph library together with HAL so we can incorporate the latest version.

Can this be made to work by adding igraph as a submodule? `#include <igraph.h>` should still work fine in this case.  See the `cmake-project-2` example here: https://github.com/igraph/igraph-example

Now I have to say that I am far from a CMake expert. igraph's build system is mostly maintained by someone else. So I may be missing something.

But if it's reasonably easy, we're willing to adapt on the igraph side to make it easier to vendor igraph sources in other libraries. I believe this _should_ work with the submodule approach already, but if there are issues, let me know.

Reply:
FYI I just re-tested the example project I suggested you use and it's not working on my machine. I am not sure why, perhaps newer CMake works differently? We'll take a look and try to fix it.

Reply:
Actually it's all fine. The submodule in my repo was in an inconsistent state. This approach should be usable. Let me know if there are problems.

Reply:
Thanks for testing @szhorvat ! We will look into the option to have igraph added as `submodule` eventually. Some time ago people were reluctant to use `submodule` since it came with very little control over the installed version and when to upgrade (or avoid upgrade due to API break) the external package - but things might have evolved since. Right now I am very happy that we finally -with your help- managed to link HAL with the latest igraph version. So I'll go ahead and merge the PR by now.

Reply:
Note that it doesn't need to be a git submodule: things should work for as long as you can get the igraph sources in a directory and use `add_subdirectory()` in CMake.

Reply:
That is what we do 
[here](https://github.com/emsec/hal/blob/master/cmake/detect_dependencies.cmake#L277)